core/src/main/java/org/infinispan/marshall/core/ExternallyMarshallable.java

```diff
@@ -2,18 +2,18 @@
 
 import java.io.Serializable;
 
-// Temporary class, should become active by default when testing
 final class ExternallyMarshallable {
 
    private ExternallyMarshallable() {
       // Static class
    }
 
-   public static boolean isAllowed(Object obj) {
+   static boolean isAllowed(Object obj) {
       String pkg = obj.getClass().getPackage().getName();
       boolean isBlackList =
             obj instanceof Serializable
             && isMarshallablePackage(pkg)
+            && !(obj instanceof ExternalPojo)
             && !isWhiteList(obj.getClass().getName());
```

Comment:
> I'm not sure how this is supposed to work for user classes

^ That was never the objective of this, but only for test classes and the check only happens when assertions are enabled.